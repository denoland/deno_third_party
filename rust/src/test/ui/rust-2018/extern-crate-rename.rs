// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

// aux-build:edition-lint-paths.rs
// run-rustfix

// Oddball: crate is renamed, making it harder for us to rewrite
// paths. We don't (and we leave the `extern crate` in place).

#![feature(rust_2018_preview)]
#![deny(absolute_paths_not_starting_with_crate)]

extern crate edition_lint_paths as my_crate;

use my_crate::foo;
//~^ ERROR absolute paths must start
//~| WARNING this was previously accepted

fn main() {
    foo();
}

