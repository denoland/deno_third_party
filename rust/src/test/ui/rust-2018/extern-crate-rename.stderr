error: absolute paths must start with `self`, `super`, `crate`, or an external crate name in the 2018 edition
  --> $DIR/extern-crate-rename.rs:22:5
   |
LL | use my_crate::foo;
   |     ^^^^^^^^^^^^^ help: use `crate`: `crate::my_crate::foo`
   |
note: lint level defined here
  --> $DIR/extern-crate-rename.rs:18:9
   |
LL | #![deny(absolute_paths_not_starting_with_crate)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in the 2018 edition!
   = note: for more information, see issue TBD

error: aborting due to previous error

